### DISCUSSION TITLE
Using Keycloak with Custom Authentication Sub-Process (or Keycloak as an SSO proxy)

### DISCUSSION THREAD

**Discussion by ronchi** [keycloak/keycloak#14822] [Q&A]
My (non-profit) organization has a very antiquated SOAP process for authentication.  It is a big organization that won't update, so we're stuck with the antiquated process.  Unfortunately, newer software services use OAuth, SAML, etc. and we are looking for a way to utilize those new services with an SSO that still employs the antiquated authentication process.   In other words, can we use KeyCloak as an SSO proxy that, during the authentication process, calls an API or runs a Python script (on the server) that would return the authentication result.  Is that possible with KeyCloak?

**Comment by sschu** [comment]
Probably not out-of-the-box, but Keycloak is extensible so you might try to create your custom extension (probably an authenticator) that does what you want.
